### PR TITLE
Improve compatibility page

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

closes https://github.com/inmanta/inmanta-core/issues/7428

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

**Comment by Hugo-Inmanta** [comment]
On hold until 08/04 for discussion about  `rename / restructure json component_requirements? Discuss on Slack`

File: docs/_templates/components_requirements.tmpl [hunk 1]
```diff
@@ -3,17 +3,27 @@
...
+The python packages are of two types: one core package and several extension packages.
 
 {% if data['component_requirements']%}
+
+The following table shows the range of compatible versions for Inmanta python components.
+These compatible versions are defined for the whole lifetime of this major version i.e.
+they are not pinned to a specific three-digit version. Upper bounds may be added when a new major version is released.
+
+This table is only meant as a reference for developers of Inmanta extensions. This information is not relevant
+when installing the orchestrator via rpm as is recommended in the :ref:`install-server` section.
```
Comment (arnaudsjs) [inline docs/_templates/components_requirements.tmpl:16]:
```suggestion
when installing the orchestrator via RPM as is recommended in the :ref:`install-server` section.
```

File: docs/_templates/components_requirements.tmpl [hunk 2]
```diff
@@ -3,17 +3,27 @@
 Inmanta core and extensions versions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+The orchestrator is constituted of several components (python packages and npm packages).
+The python packages are of two types: one core package and several extension packages.
 
 {% if data['component_requirements']%}
+
+The following table shows the range of compatible versions for Inmanta python components.
```
Comment (arnaudsjs) [inline docs/_templates/components_requirements.tmpl:11]: I think we should make it clear that the production version is the starting point and from that we define version constraints on the versions of the components.
```suggestion
The following table shows the ranges of the Inmanta Python components, this product version is compatible with.
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: docs/_templates/components_requirements.tmpl [hunk 3]
```diff
@@ -3,17 +3,27 @@
...
+These compatible versions are defined for the whole lifetime of this major version i.e.
```
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply docs/_templates/components_requirements.tmpl:12]: Based on Sander his comment on Slack I think we need to change the following here: 

* I would change the sentence "These compatible versions are defined for the whole lifetime of this major version i.e. they are not pinned to a specific three-digit version." to "the lower bound of these version constraints will remain the same for the whole lifetime of this major version". The latter is just a lot more clear to me.
* We also need to mention explicitly that these version constraints will only result in compatible versions of they are used as input towards a dependency resolver like pip. The current explanation gives the impression that if your installed packages satisfy these constraints, they are compatible with each other. But that's not necessarily true.

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**